### PR TITLE
testing/cmocka: update cmocka version and add patch

### PR DESCRIPTION
**Pull request by yintao707**
Specify cmocka as the latest version 1.15
add patch to enable cmocka to output with syslog

## Summary
Update cmocka version and add patch
## Impact
None
## Testing
Pass vela CI

### CODE REVIEW

File: testing/cmocka/.gitignore [hunk 1]
```diff
@@ -1 +0,0 @@
-/cmocka
```
Comment (xiaoxiang781216) [inline testing/cmocka/.gitignore:1]: shuld we keep this?

**Comment by acassis** [comment]
@yintao707 maybe these modifications could be submitted to mainline of the cmocka project

**Comment by xiaoxiang781216** [comment]
They are here:
https://gitlab.com/cmocka/cmocka/-/merge_requests/59
https://gitlab.com/cmocka/cmocka/-/merge_requests/58

File: testing/cmocka/.gitignore (hunk 1 above)
Comment (yintao707) [reply testing/cmocka/.gitignore]: Done,thankyou!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**